golang/internal/helper/docker/container.go

```diff
@@ -102,32 +118,34 @@ func RemoveContainer(ctx context.Context, dog *dogger.DeploymentLogger, nameFilt
 }
 
 // Check the existence of containers, then return it
-func GetAllContainersByName(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string) ([]types.Container, error) {
+func GetAllContainersByName(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string) ([]*types.Container, error) {
```

Comment:
What's the point of returning `nil` containers? Why not just leave them out of the array?

Reply:
This is an optimization for the foreaches